dbt_semantic_interfaces/parsing/where_filter/parameter_set_factory.py

```diff
@@ -87,14 +87,17 @@ def create_dimension(dimension_name: str, entity_path: Sequence[str] = ()) -> Di
     @staticmethod
     def create_entity(entity_name: str, entity_path: Sequence[str] = ()) -> EntityCallParameterSet:
         """Gets called by Jinja when rendering {{ Entity(...) }}."""
-        group_by_item_name = DunderedNameFormatter.parse_name(entity_name)
-        if len(group_by_item_name.entity_links) > 0 or group_by_item_name.time_granularity is not None:
+        structured_dundered_name = DunderedNameFormatter.parse_name(entity_name)
+        if structured_dundered_name.time_granularity is not None:
```

Comment:
Curious why we don't allow passing granularity in the where filter?

Reply:
We want to eliminate `entity__dimension__time_granularity` formats altogether, because it causes a ton of confusion for us and requires theoretically unbounded future restrictions on user names for things - like today you can't call a date column `day`. I mean, you shouldn't, but companies totally do this. Really, they can't call things `extract_day`, either, but I suspect that's a rare corner case.

Ideally, for input, the dunder will only ever establish a link between a primary entity and some attribute it points to (dimension, entity, whatever), rather than being "that thing that glues together a bunch of arbitrary stuff that we can't distinguish from user-land names for things."